examples/cms/src/App.tsx

```diff
@@ -0,0 +1,142 @@
+import { framer } from "framer-plugin"
+import "./App.css"
+import { useLayoutEffect, useState } from "react"
+import {
+    computeFieldConfigs,
+    DataSource,
+    FieldConfig,
+    getDataSources,
+    listDataSourcesIds,
+    PLUGIN_COLLECTION_SYNC_REFERENCE_KEY,
+    PLUGIN_COLLECTION_SYNC_SLUG_KEY,
+    syncCollection,
+} from "./data"
+import { FieldMapping } from "./FieldMapping"
+
+const activeCollection = await framer.getManagedCollection()
+const existingFields = activeCollection ? await activeCollection.getFields() : []
+
+const syncDataSourceId = await activeCollection.getPluginData(PLUGIN_COLLECTION_SYNC_REFERENCE_KEY)
+const syncSlugFieldId = await activeCollection.getPluginData(PLUGIN_COLLECTION_SYNC_SLUG_KEY)
+
+const syncDataSource = syncDataSourceId ? await getDataSources(syncDataSourceId) : null
+
+let savedFieldsConfig: FieldConfig[] | undefined
+
+if (syncDataSource) {
+    savedFieldsConfig = computeFieldConfigs(existingFields, syncDataSource)
+}
+
+if (framer.mode === "syncManagedCollection" && savedFieldsConfig && syncDataSource && syncSlugFieldId) {
+    try {
+        await syncCollection(
+            syncDataSource,
+            savedFieldsConfig.filter(field => field.field && !field.isNew),
+            syncSlugFieldId
+        )
+        await framer.closePlugin(`Synchronization successful`, {
+            variant: "success",
+        })
+    } catch (error) {
+        console.error(error)
+        framer.closePlugin(`Failed to sync collection`, {
+            variant: "error",
+        })
+    }
+}
```

Comment:
Shouldn't all of these things only happen when the mode is `syncManagedCollection`? Otherwise its just going to delay the app from showing up, right?

Reply:
I think it would be much better to move all this into an async function

Reply:
And then the result of the function could be a string that says something like:

```ts
async function syncExistingCollection(): Promise<"success" | "needsConfiguration">
```